uwblueprint/feeding-canadian-kids
# Create Onboarding Request

## Notion ticket link

[API: Create onboarding request](https://app.zenhub.com/workspaces/fck-627480c10d8974001872b0ef/issues/uwblueprint/feeding-canadian-kids/2)



## Implementation description
* see models here: https://www.notion.so/uwblueprintexecs/Models-ee57721ed8be42e0ab63254acb1e6439#54edb83dabf24002926f5001f15a4ece



## Steps to test
1. Go to localhost:5000/graphql
2. Then input and verify that it is a success and date_submitted makes sense.

```
mutation testCreateOnboardingRequest {
  createOnboardingRequest(
    userInfo: 
    {contactName: "Jessie Peng",
     contactEmail: "jpeng@gmail.com",
     contactPhone: "647641", 
     role: "Admin"
    }
  ) {
    onboardingRequest {
      id
      info {
        contactName,
        contactEmail,
        contactPhone,
        role
      }
      dateSubmitted
      status
    }
  }
}
```

Also test that this throws an error (incorrect role given):

```
mutation testCreateOnboardingRequest {
  createOnboardingRequest(
    userInfo: 
    {contactName: "Jessie Peng",
     contactEmail: "jpeng@gmail.com",
     contactPhone: "647641", 
     role: "Incorrect Role"
    }
  ) {
    onboardingRequest {
      id
      info {
        contactName,
        contactEmail,
        contactPhone,
        role
      }
      dateSubmitted
      status
    }
  }
}
```
4. Also run` docker exec -it fck_backend /bin/bash -c "pip install -e . && pytest"`and verify that the tests test_models.py and test_onboarding_request.py pass.



## What should reviewers focus on?
* 


## Checklist
